bindgen/ir/var.rs

```diff
@@ -164,18 +164,32 @@ fn handle_function_macro(
 ) {
     let is_closing_paren = |t: &ClangToken| {
         // Test cheap token kind before comparing exact spellings.
-        t.kind == clang_sys::CXToken_Punctuation && t.spelling() == b")"
+        t.kind == clang_sys::CXToken_Punctuation &&
+            t.spelling().to_bytes() == b")"
     };
     let tokens: Vec<_> = cursor.tokens().iter().collect();
     if let Some(boundary) = tokens.iter().position(is_closing_paren) {
-        let mut spelled = tokens.iter().map(ClangToken::spelling);
-        // Add 1, to convert index to length.
-        let left = spelled.by_ref().take(boundary + 1);
-        let left = left.collect::<Vec<_>>().concat();
-        if let Ok(left) = String::from_utf8(left) {
-            let right: Vec<_> = spelled.collect();
-            callbacks.func_macro(&left, &right);
-        }
+        let mut tokens = tokens
+            .iter()
+            .map(|token| token.spelling().to_str().unwrap())
+            .collect::<Vec<_>>();
```

Comment:
Same questions, I'd rather avoid crashing for non-utf8 input and just expose bytes?

Reply:
This is simply extracted from https://github.com/rust-lang/rust-bindgen/pull/2369 in order to be able to use `cmacro`.

`cmacro` needs to assume a single encoding to correctly parse macros. The same is true for users that implement `ParseCallbacks::fn_macro`, they cannot just *guess* the encoding without knowing which file the byte slice came from.

I guess we can just skip macros containing invalid UTF-8 for now, though.